final_task/rss_reader/news_converter.py

```diff
@@ -103,3 +99,100 @@ def item_text_with_imgs2html(text, img_links):
 
     text_and_imgs += text
     return text_and_imgs
+
+
+def news2pdf(item_groups, file_path):
+    """ Write news in PDF file
+
+    :type item_groups: list of 'item_group.ItemGroup'
+    :type file_path: str
+    """
+    width = 180
+
+    pdf = FPDF()
+    pdf.add_page()
+
+    pdf.add_font('DejaVu', '', 'fonts/DejaVuSansCondensed.ttf', uni=True)
+    pdf.add_font('DejaVuBold', '', 'fonts/DejaVuSansCondensed-Bold.ttf', uni=True)
+    pdf.add_font('DejaVuOblique', '', 'fonts/DejaVuSansCondensed-Oblique.ttf', uni=True)
+
+    num = 0
+
+    for item_gr in item_groups:
+        pdf.set_font('DejaVuBold', size=24)
+        pdf.set_text_color(0, 10, 180)
+        pdf.multi_cell(width, 260, item_gr.feed, align='C')
+
+        for item in item_gr.items:
+            pdf.add_page()
+
+            pdf.set_font('DejaVuBold', size=18)
+            pdf.set_text_color(0, 0, 0)
+            pdf.multi_cell(width, 16, item.title, align='C')
+
+            pdf.set_font('DejaVu', size=16)
+
+            text = item.text
+            for ind, link in enumerate(item.img_links):
+                alt, before_picture, text = parse_item_text(text, ind+1)
+
+                if before_picture:
+                    pdf.multi_cell(width, 16, before_picture)
+
+                try:
+                    img = requests.get(link)
+                    if img.status_code != 200 or imghdr.what(None, img.content) != 'jpeg':
+                        raise requests.exceptions.ConnectionError
```

Comment:
It is better to raise exception instances instead of exception class objects

Reply:
fixed